x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared/data_tier_allocation_field/data_tier_allocation_field.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiDescribedFormGroup, EuiFormRow, EuiSpacer } from '@elastic/eui';
+
+import { useKibana, useFormData } from '../../../../../../../shared_imports';
+
+import { PhaseWithAllocation } from '../../../../../../../../common/types';
+
+import { getAvailableNodeRoleForPhase } from '../../../../../../lib/data_tiers';
+
+import { isNodeRoleFirstPreference } from '../../../../../../lib';
+
+import {
+  DataTierAllocation,
+  DefaultAllocationNotice,
+  NoNodeAttributesWarning,
+  NodesDataProvider,
+  CloudDataTierCallout,
+} from './components';
+
+const i18nTexts = {
+  title: i18n.translate('xpack.indexLifecycleMgmt.common.dataTier.title', {
+    defaultMessage: 'Data allocation',
+  }),
+};
+
+interface Props {
+  phase: PhaseWithAllocation;
+  description: React.ReactNode;
+}
+
+/**
+ * Top-level layout control for the data tier allocation field.
+ */
+export const DataTierAllocationField: FunctionComponent<Props> = ({ phase, description }) => {
+  const {
+    services: { cloud },
+  } = useKibana();
+
+  const dataTierAllocationTypePath = `_meta.${phase}.dataTierAllocationType`;
+  const [formData] = useFormData({ watch: dataTierAllocationTypePath });
+  const allocationType = get(formData, dataTierAllocationTypePath);
+
+  return (
+    <NodesDataProvider>
+      {({ nodesByRoles, nodesByAttributes, isUsingDeprecatedDataRoleConfig }) => {
+        const hasNodeAttrs = Boolean(Object.keys(nodesByAttributes ?? {}).length);
+
+        const renderNotice = () => {
+          switch (allocationType) {
+            case 'node_roles':
+              const isCloudEnabled = cloud?.isCloudEnabled ?? false;
```

Comment:
nit: IMO this type of check should be done only once at `start` and persisted for all the consumers.